src/mono/wasm/runtime/binding_support.js

```diff
@@ -88,40 +88,54 @@ var BindingSupportLib = {
 			if (!wasm_runtime_class)
 				throw "Can't find " + binding_fqn_class + " class";
 
-			var get_method = function(method_name) {
-				var res = BINDING.find_method (wasm_runtime_class, method_name, -1)
-				if (!res)
-					throw "Can't find method " + namespace + "." + classname + ":" + method_name;
-				return res;
+			this.runtime = {
+				class: wasm_runtime_class
+			};
+
+			var bind_method = function(method_name, signature, namespace = BINDING.runtime) {
+				var method = BINDING.find_method (namespace.class, method_name, -1);
+				namespace[method_name ] = function () {
+					try {
+						args = [];
+						Array.prototype.forEach.call(arguments, function (el) {
+							args.push (el);
+						});
+						return BINDING.call_method (method, null, signature, args);
+					} catch (ex) {
+						console.error (`unable to call method: ${method_name} args: ${args.toString()}`, ex.stack);
+					}
+				}
+				return namespace[method_name];
 			}
-			this.bind_js_obj = get_method ("BindJSObject");
-			this.bind_core_clr_obj = get_method ("BindCoreCLRObject");
-			this.bind_existing_obj = get_method ("BindExistingObject");
-			this.unbind_raw_obj_and_free = get_method ("UnBindRawJSObjectAndFree");			
-			this.get_js_id = get_method ("GetJSObjectId");
-			this.get_raw_mono_obj = get_method ("GetDotNetObject");
-
-			this.box_js_int = get_method ("BoxInt");
-			this.box_js_double = get_method ("BoxDouble");
-			this.box_js_bool = get_method ("BoxBool");
-			this.is_simple_array = get_method ("IsSimpleArray");
-			this.setup_js_cont = get_method ("SetupJSContinuation");
-
-			this.create_tcs = get_method ("CreateTaskSource");
-			this.set_tcs_result = get_method ("SetTaskSourceResult");
-			this.set_tcs_failure = get_method ("SetTaskSourceFailure");
-			this.tcs_get_task_and_bind = get_method ("GetTaskAndBind");
-			this.get_call_sig = get_method ("GetCallSignature");
-
-			this.object_to_string = get_method ("ObjectToString");
-			this.get_date_value = get_method ("GetDateValue");
-			this.create_date_time = get_method ("CreateDateTime");
-			this.create_uri = get_method ("CreateUri");
-
-			this.safehandle_addref = get_method ("SafeHandleAddRef");
-			this.safehandle_release = get_method ("SafeHandleRelease");
-			this.safehandle_get_handle = get_method ("SafeHandleGetHandle");
-			this.safehandle_release_by_handle = get_method ("SafeHandleReleaseByHandle");
+
+			bind_method ("BindJSObject", "iii");
+			bind_method ("BindCoreCLRObject", "ii")
+			bind_method ("BindExistingObject", "mi");
+			bind_method ("UnBindRawJSObjectAndFree", "ii");
+			bind_method ("GetJSObjectId", "m");
+			bind_method ("GetDotNetObject", "im");
+
+			bind_method ("BoxInt", "im");
+			bind_method ("BoxDouble", "dm");
+			bind_method ("BoxBool", "im");
+			bind_method ("IsSimpleArray", "mi");
+			bind_method ("SetupJSContinuation", "mo")
+
+			bind_method ("CreateTaskSource", "i");
+			bind_method ("SetTaskSourceFailure", "oo");
```

Comment:
`s,SetTaskSourceFailure,SetTaskSourceResult,`

Reply:
good catch, clearly we don't have tests for this